repo2docker/contentproviders/doi.py

```diff
@@ -49,7 +49,9 @@ def doi2url(self, doi):
             doi = normalize_doi(doi)
 
             try:
-                resp = self._request(f"https://doi.org/{doi}")
+                # We don't need to fetch the *contents* of the page the doi resolves to -
+                # only need to know what it redirects to.
+                resp = self._request(f"https://doi.org/{doi}", allow_redirects=False)
```

Comment:
Are there any rules on whether a DOI can redirect multiple times?

Reply:
@manics great question that led me down a rabbithole! I'm now convinced this approach is not robust enough, and will open a different series of PRs - starting with https://github.com/jupyterhub/repo2docker/pull/1389, and https://github.com/jupyterhub/repo2docker/pull/1390 is a WIP